### DISCUSSION TITLE
Third party payment gateway cartridges not working in SFRA architecture

### DISCUSSION THREAD

**Discussion by rajanbhuyan** [SalesforceCommerceCloud/sfcc-ci#423] [Team Posts]
I was trying to set up third-party payment gateways like PayPal, Adyen, etc. in SFRA architecture. For that, I followed all the steps provided in the installation documents of PayPal and Adyen and the gateways are also enabled in the Business Manager. Unfortunately, PayPal and Adyen are not showing up in the front end - the "RefArch" site. I have cross-checked many times the documents provided by the vendor to make sure I am doing the right thing. So, I am bit concerned if I am missing anything or there is something different way to set it up in SFRA. Just a note that these gateways worked perfectly in the SGJC implementation.

Any help will be highly appreciated!!

Thanks

**Comment by zlatinz** [comment]
Hi @rajanbhuyan ,

Usually from the information in the marketplace, e.g. https://www.salesforce.com/products/commerce-cloud/partner-marketplace/partners/adyen/ 
you can click on "Download Integration" which leads you to the LINK cartridge information in xchange.
e.g. https://xchange.demandware.com/docs/DOC-10607

From there in the "Download Information" section you have two very important elements:

"Integration & Documentation" which gives you information about the version number (e.g. Release 18.3.0) and allows you to download the actual LINK cartridge.
-and-
"Feedback & Questions" which gives you the email address you can use to ask for help with that particular LINK cartridge integration, if you are certain the issue is missing configuration step in the documentation or something (partnerships@adyen.com)

**Comment by shubhamgoel710** [comment]
Hi,
If you're having trouble with PayPal button loading, it might be possible because the JS is executing before the DOM is built. To fix that, look at the file scripts.isml in the int_paypal_sfra cartridge, and add 'defer' attribute to script tags.
Hope this resolves your problem. 

Thanks
Shubham

**Comment by rpitla** [comment]
Hi @rajan

Issue might be also that these cartridges are not compatible with the latest SFRA version 4.2.1. All Payment LINK Cartridges had to be updated after SFRA v4.0.0 since it came with a payment hook changes.
Adyen and PayPal are currently working on updating their Cartridges and they're in the queue to get certified with the latest SFRA version.
Their existing Cartridges should work fine with SFRA v3.6.0.

Hope this helps!
Rama

**Comment by rajanbhuyan** [comment]
Thanks @rpitla ! 
I changed the SFRA version still no luck. None of the payment gateways are showing up on the Payments Page.

**Comment by rajanbhuyan** [comment]
> Hi,
> If you're having trouble with PayPal button loading, it might be possible because the JS is executing before the DOM is built. To fix that, look at the file scripts.isml in the int_paypal_sfra cartridge, and add 'defer' attribute to script tags.
> Hope this resolves your problem.
> 
> Thanks
> Shubham

I tried this as well but no luck!

**Comment by vkasthuri77** [comment]
Hi , I had the same problem and I had to make a small code change in paypalMFRA.js where the buttons are getting initialized. Root cause seems to be , JS is getting executed before the DOM is built because the main function is an self-executing function where the buttons are getting initialized. Below line of code is what you will see in this JS file. 

(function () {
}());

I just enforced this piece of code to get executed after the document ready by changing as below:

(function () {
	$(function() {
	}); 	
}());
This means you will have to change the OOTB plugin cartridge as I dont see it as an common issue reported by other users so that its handled in their original version itself. As an workaround, we can override this JS file in our custom cartridge and place it before the cartridge path so that original link cartridge is untouched. Hope this helps. 
Thanks
Venkatesh
